optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/GizmoMemberAccessorEntityEnhancer.java

```diff
@@ -545,6 +556,40 @@ public static String generateKieRuntimeBuilder(ClassOutput classOutput, SolverCo
                     transformers.produce(new BytecodeTransformerBuildItem(entityClass.getName(),
                             (className, classVisitor) -> new OptaPlannerDroolsReactiveClassVisitor(entityClass, classVisitor)));
                 }
+            } else if (config.getScoreDirectorFactoryConfig().getConstraintProviderClass() != null
+                    && constraintStreamImplType == ConstraintStreamImplType.DROOLS) {
+                methodCreator = classCreator.getMethodCreator(MethodDescriptor.ofMethod(OptaPlannerDroolsInitializer.class,
+                        "setup", void.class, ScoreDirectorFactoryConfig.class));
+                ResultHandle constraintProvider = methodCreator.newInstance(
+                        MethodDescriptor.ofConstructor(config.getScoreDirectorFactoryConfig().getConstraintProviderClass()));
+                ResultHandle entityClassList = methodCreator.newInstance(MethodDescriptor.ofConstructor(ArrayList.class));
+                for (Class<?> entityClass : config.getEntityClassList()) {
+                    ResultHandle entityClassResultHandle = methodCreator.loadClass(entityClass);
+                    methodCreator.invokeInterfaceMethod(
+                            MethodDescriptor.ofMethod(List.class, "add", boolean.class, Object.class),
+                            entityClassList, entityClassResultHandle);
+                }
+                ResultHandle solutionDescriptor = methodCreator.invokeStaticMethod(
+                        MethodDescriptor.ofMethod(SolutionDescriptor.class, "buildSolutionDescriptor",
+                                SolutionDescriptor.class, DomainAccessType.class, Class.class, Map.class, Map.class,
+                                List.class),
+                        methodCreator.load(DomainAccessType.REFLECTION),
+                        methodCreator.loadClass(config.getSolutionClass()),
+                        methodCreator.loadNull(),
+                        methodCreator.loadNull(),
+                        entityClassList);
+                ResultHandle isDroolsAlphaNetworkCompilationEnabled =
+                        methodCreator.load(config.getScoreDirectorFactoryConfig().isDroolsAlphaNetworkCompilationEnabled());
+                ResultHandle kieBaseDescriptor = methodCreator.invokeStaticMethod(
+                        MethodDescriptor.ofMethod(DroolsConstraintStreamScoreDirectorFactory.class, "buildKieBase",
+                                "org.optaplanner.core.impl.score.director.stream.KieBaseDescriptor",
```

Comment:
I think we can avoid this if we make `KieBaseDescriptor` public.
We can do that, because it is in a package which we consider internal, and therefore not give backwards compatibility guarantees.